### PR TITLE
Fix - Multivalued fields in %values lists

### PR DESCRIPTION
**Pull request by wetzelj**
# Description

Related issues: #174 

Checked value before attempting to add it to values list.  If the value is MultiValue, use set.Update() instead of set.Add().

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project

# Open questions

Questions that require more discussion or to be addressed in future development:
None

### CODE REVIEW

File: deid/dicom/groups.py [hunk 1]
```diff
@@ -55,7 +56,10 @@ def extract_values_list(dicom, actions, fields=None):
         if action["action"] == "FIELD":
             for uid, field in subset.items():
                 if field.element.value not in ["", None]:
-                    values.add(field.element.value)
+                    if type(field.element.value) is MultiValue:
```
Comment (vsoch) [inline deid/dicom/groups.py:59]: Let's use isinstance here, it's a bit faster: https://switowski.com/blog/type-vs-isinstance
```suggestion
                    if isinstance(field.element.value, MultiValue):
```

File: deid/dicom/groups.py [hunk 2]
```diff
@@ -55,7 +56,10 @@ def extract_values_list(dicom, actions, fields=None):
...
+                        values.update(field.element.value)
```
Comment (vsoch) [inline deid/dicom/groups.py:60]: So just so I understand - when we have a MultiValue it is a dictionary?

**Review by vsoch — COMMENTED** [review]
This looks great - a few quick comments / questions! Is there an easy test case for this? You do a fantastic job writing tests so if the answer is no, I'm okay with taking your word that you tested this on some real data and it fixed the bug.

File: deid/dicom/groups.py (hunk 2 above)
Comment (wetzelj) [reply deid/dicom/groups.py:60]: When the value is [MultiValue](https://pydicom.github.io/pydicom/dev/reference/generated/pydicom.multival.MultiValue.html), the value in field.element.value is a "list of items that are all the same type", not a single element that we're adding into the values set.   

In the specific scenario we encountered, this was with the OtherPatientIDs field.  In all but two studies that we've deidentified using pydicom/deid, the OtherPatientIDs contained a single patient identifier, however with the two outlier studies, the modality actually sent multiple patient identifiers within this value - the MultiValue value was a list of strings.

**Comment by wetzelj** [comment]
> This looks great - a few quick comments / questions! Is there an easy test case for this? You do a fantastic job writing tests so if the answer is no, I'm okay with taking your word that you tested this on some real data and it fixed the bug.

I didn't create a test case for this because I would have had to fabricate multi-occurrence within the test dicom image and given the nature of this change, questioned if it was really worth it.  If you want me to, I can come up with something for it...

File: deid/dicom/groups.py (hunk 2 above)
Comment (vsoch) [reply deid/dicom/groups.py:60]: Ahh gotcha, so you are essentially doing:

```python
testset= set()
testset.add('value1')
testset.update(['value2', 'value3'])
{'value1', 'value2', 'value3'}
```

**Comment by vsoch** [comment]
Nah I think we are good! If you are going to do some other PRs soon then I'll merge here (but not release) and we can release an updated version when all these changes are in. As always, thanks you @wetzelj !

File: deid/dicom/groups.py (hunk 2 above)
Comment (wetzelj) [reply deid/dicom/groups.py:60]: Exactly.

**Review by vsoch — APPROVED** [review]

**Pull request merged**